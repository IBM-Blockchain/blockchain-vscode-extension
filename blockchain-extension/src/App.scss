@import 'node_modules/carbon-components/scss/components/accordion/_accordion';
@import 'node_modules/carbon-components/scss/components/button/_button';
@import 'node_modules/carbon-components/scss/components/link/link';
