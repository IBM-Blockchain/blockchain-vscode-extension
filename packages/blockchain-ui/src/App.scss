@import '@carbon/colors/scss/colors';

#root{
    .app-container__tutorial-page {
        background-color: $gray-10;
        position: absolute;
        width: 100%;
        z-index: -1;
        box-shadow: inset 0px 164x #FFFFFF;
        top: 0;
        left: 0;

    }
}
