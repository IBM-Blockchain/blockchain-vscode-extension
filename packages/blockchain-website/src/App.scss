.title {
  margin-top: 8px;
  margin-bottom: 8px;
  font-size: 20px;
  color: #161616;
  letter-spacing: 0;
  line-height: 26px;
}

.app-container {
  margin-left: 112px;
}

.issues-container {
  max-width: 1244px;
}

.upcoming-issues-container {
  padding-bottom: 32px; 
}

.status {
  margin-top: 40px; 
  font-size: 28px;
  color: #161616;
  letter-spacing: 0;
  height: 20px;
}

.description {
  max-width: 608px;
  font-size: 12px;
  color: #525252;
  letter-spacing: 0.32px;
  text-align: left;
  line-height: 16px;
}

.heading-container {
  display: flex;
  flex-flow: column;
}

.header-bar {
  background-color: #202123;
  display: flex;
  flex-flow: column;
  max-height: 48px;
}

.ibm {
  font-family: 'IBMPlexSans', sans-serif;
  font-size: 14px;
  color: #FFFFFF;
  letter-spacing: 0.1px;
  text-align: left;
  line-height: 18px;
  padding-top: 14px;
  padding-bottom: 16px;
  padding-right: 3px;
  padding-left: 16px;
}

.page-title {
  font-family: 'IBMPlexSans-SemiBold', sans-serif;
  font-size: 14px;
  color: #FFFFFF;
  letter-spacing: 0.1px;
  text-align: left;
  line-height: 18px;
  padding-top: 14px;
  padding-bottom: 16px;
}