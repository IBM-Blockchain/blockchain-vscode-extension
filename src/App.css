.panel {
  margin-bottom: 20px;
  background-color: #fff;
  border: 1px solid transparent;
  border-radius: 4px;
  -webkit-box-shadow: 0 1px 1px rgba(0, 0, 0, .05);
          box-shadow: 0 1px 1px rgba(0, 0, 0, .05);
}
.panel-body {
  padding: 15px;
}
.panel-heading {
  padding: 10px 15px;
  border-bottom: 1px solid transparent;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
}
.panel-heading > .dropdown .dropdown-toggle {
  color: inherit;
}
.panel-title {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: inherit;
}
.panel-bug {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: #d73a4a;
}
.panel-feature {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: #a2eeef;
}
.panel-success {
  border-color: #d6e9c6;
}
.panel-success > .panel-heading {
  color: #3c763d;
  background-color: #dff0d8;
  border-color: #d6e9c6;
}
.panel-success > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #d6e9c6;
}
.panel-success > .panel-heading .badge {
  color: #dff0d8;
  background-color: #3c763d;
}
.panel-success > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #d6e9c6;
}
.panel-danger {
  border-color: #ebccd1;
}
.panel-danger > .panel-heading {
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
}
.panel-danger > .panel-heading + .panel-collapse > .panel-body {
  border-top-color: #ebccd1;
}
.panel-danger > .panel-heading .badge {
  color: #f2dede;
  background-color: #a94442;
}
.panel-danger > .panel-footer + .panel-collapse > .panel-body {
  border-bottom-color: #ebccd1;
}

hr {
  height: 0;
  -webkit-box-sizing: content-box;
     -moz-box-sizing: content-box;
          box-sizing: content-box;
}

.label,
.badge {
  background-color: #b5b5b5;
  color: #fff;
  font-size: 12px;
  font-size: 0.75rem;
  line-height: 2;
  font-weight: bold;
  padding: 3px 7px;
  text-align: center;
  vertical-align: baseline;
  white-space: nowrap;
}

.label {
  border-radius: 4px;
  display: inline;
}

.label, .badge {
  background-color: #b5b5b5;
  color: #fff;
  font-size: 12px;
  font-size: 0.75rem;
  line-height: 2;
  font-weight: bold;
  padding: 3px 7px;
  text-align: center;
  vertical-align: baseline;
  white-space: nowrap;
}

.bg-bug {
  background-color: #d73a4a !important;
}

.bg-feature {
  background-color: #02ccbf !important;
}

.green{
  background:green;
}

.red{
  background:red;
}

.grey{
  background:gray;
}

.panel-danger .panel-heading h3,
.panel-danger .panel-body p
{
  font-weight:600;
}

h2 a{
  font-weight:300;
  color:#aaa;
}

footer{
  margin-top:20px;
  margin-bottom:20px;
}

.logo {
  width: 35px;
  margin-top: -6px;
}

@media (min-width: 992px) {
  .col-md-1, .col-md-2, .col-md-3, .col-md-4, .col-md-5, .col-md-6, .col-md-7, .col-md-8, .col-md-9, .col-md-10, .col-md-11, .col-md-12 {
    float: left;
  }
  .col-md-12 {
    width: 100%;
  }
  .col-md-11 {
    width: 91.66666667%;
  }
  .col-md-10 {
    width: 83.33333333%;
  }
  .col-md-9 {
    width: 75%;
  }
  .col-md-8 {
    width: 66.66666667%;
  }
  .col-md-7 {
    width: 58.33333333%;
  }
  .col-md-6 {
    width: 50%;
  }
  .col-md-5 {
    width: 41.66666667%;
  }
  .col-md-4 {
    width: 33.33333333%;
  }
  .col-md-3 {
    width: 25%;
  }
  .col-md-2 {
    width: 16.66666667%;
  }
  .col-md-1 {
    width: 8.33333333%;
  }
  .col-md-pull-12 {
    right: 100%;
  }
  .col-md-pull-11 {
    right: 91.66666667%;
  }
  .col-md-pull-10 {
    right: 83.33333333%;
  }
  .col-md-pull-9 {
    right: 75%;
  }
  .col-md-pull-8 {
    right: 66.66666667%;
  }
  .col-md-pull-7 {
    right: 58.33333333%;
  }
  .col-md-pull-6 {
    right: 50%;
  }
  .col-md-pull-5 {
    right: 41.66666667%;
  }
  .col-md-pull-4 {
    right: 33.33333333%;
  }
  .col-md-pull-3 {
    right: 25%;
  }
  .col-md-pull-2 {
    right: 16.66666667%;
  }
  .col-md-pull-1 {
    right: 8.33333333%;
  }
  .col-md-pull-0 {
    right: auto;
  }
  .col-md-push-12 {
    left: 100%;
  }
  .col-md-push-11 {
    left: 91.66666667%;
  }
  .col-md-push-10 {
    left: 83.33333333%;
  }
  .col-md-push-9 {
    left: 75%;
  }
  .col-md-push-8 {
    left: 66.66666667%;
  }
  .col-md-push-7 {
    left: 58.33333333%;
  }
  .col-md-push-6 {
    left: 50%;
  }
  .col-md-push-5 {
    left: 41.66666667%;
  }
  .col-md-push-4 {
    left: 33.33333333%;
  }
  .col-md-push-3 {
    left: 25%;
  }
  .col-md-push-2 {
    left: 16.66666667%;
  }
  .col-md-push-1 {
    left: 8.33333333%;
  }
  .col-md-push-0 {
    left: auto;
  }
  .col-md-offset-12 {
    margin-left: 100%;
  }
  .col-md-offset-11 {
    margin-left: 91.66666667%;
  }
  .col-md-offset-10 {
    margin-left: 83.33333333%;
  }
  .col-md-offset-9 {
    margin-left: 75%;
  }
  .col-md-offset-8 {
    margin-left: 66.66666667%;
  }
  .col-md-offset-7 {
    margin-left: 58.33333333%;
  }
  .col-md-offset-6 {
    margin-left: 50%;
  }
  .col-md-offset-5 {
    margin-left: 41.66666667%;
  }
  .col-md-offset-4 {
    margin-left: 33.33333333%;
  }
  .col-md-offset-3 {
    margin-left: 25%;
  }
  .col-md-offset-2 {
    margin-left: 16.66666667%;
  }
  .col-md-offset-1 {
    margin-left: 8.33333333%;
  }
  .col-md-offset-0 {
    margin-left: 0;
  }
}