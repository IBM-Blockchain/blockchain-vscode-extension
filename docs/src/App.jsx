import React, { Component } from 'react';
import 'bootstrap/scss/bootstrap.scss';
import logo from './ext_logo.svg';
import StatusList from './components/StatusList/StatusList';
import UpcomingList from './components/UpcomingList/UpcomingList';
import './App.scss';
import CurrentVersion from './components/CurrentVersion/CurrentVersion';


class App extends Component {
  render() {
    return (
        <div className="container">
            <div className="row">
                <div>
                    <h1 className="text-muted"><img className="logo" src={logo} alt='IBM Blockchain Logo'></img> IBM Blockchain Platform - Status Page</h1>
                    <hr />
                    <h2 className='text-muted'> <CurrentVersion></CurrentVersion> </h2>
                    <p className="lead">
                        Check this page for information on any current incidents effecting the latest version of the IBM Blockchain Platform extension for Visual Studio Code, as well as planned content for the next release.
                    </p>
                    <StatusList></StatusList>
                    <hr/>
                    <UpcomingList></UpcomingList>
                </div>
            </div>
        </div>
        )
    }
}

export default App;
