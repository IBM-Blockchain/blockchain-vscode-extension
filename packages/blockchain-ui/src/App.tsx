import * as React from 'react';
import './App.scss';
import { HashRouter as Router, Route, Redirect } from 'react-router-dom';
import TransactionHome from './components/TransactionHome/TransactionHome';
import TransactionCreate from './components/TransactionCreate/TransactionCreate';
import ISmartContract from './interfaces/ISmartContract';
import Utils from './Utils';

interface AppState {
    redirectPath: string;
    gatewayName: string;
    activeSmartContract: ISmartContract;
    smartContracts: Array<ISmartContract>;
    transactionOutput: string;
}

class App extends React.Component<{}, AppState> {
    constructor(props: {}) {
        super(props);
        this.state = {
            redirectPath: '',
            gatewayName: '',
            activeSmartContract: {
                name: '',
                version: '',
                channel: '',
                label: '',
                transactions: [],
                namespace: ''
            },
            smartContracts : [],
            transactionOutput: ''
        };
        this.switchSmartContract = this.switchSmartContract.bind(this);
        this.postMessageHandler = this.postMessageHandler.bind(this);
    }

    componentDidMount(): void {
        window.addEventListener('message', (event: MessageEvent) => {
            if (event.data.output) {
                this.setState({
                    transactionOutput: event.data.output
                });
            } else {
                this.setState({
                    redirectPath: event.data.path,
                    gatewayName: event.data.state ? event.data.state.gatewayName : this.state.gatewayName,
                    activeSmartContract: event.data.state ? event.data.state.activeSmartContract : this.state.activeSmartContract,
                    smartContracts: event.data.state ? event.data.state.smartContracts : this.state.smartContracts,
                });
            }
        });
    }

    switchSmartContract(newActiveContractLabel: string): void {
        const smartContracts: Array<ISmartContract> = this.state.smartContracts;
        this.setState({
            activeSmartContract: smartContracts.find((obj: ISmartContract) => obj.label === newActiveContractLabel) as ISmartContract,
            smartContracts: smartContracts
        });
    }

    postMessageHandler(command: string, data?: any): void {
        if (data === undefined) {
            data = {
                gatewayName: this.state.gatewayName,
                activeSmartContract: this.state.activeSmartContract,
                smartContracts: this.state.smartContracts
            };
        }

        Utils.postToVSCode({
            command: command,
            data: data
        });
    }

    render(): JSX.Element {
        if (this.state.redirectPath === '') {
            return <div></div>;
        } else {
            return (
                <Router>
                    <div>
                        <Route render={(): JSX.Element => <Redirect push to={this.state.redirectPath}/>}></Route>
                        <Route exact path='/transaction' render={(): JSX.Element =>
                            <TransactionHome gatewayName={this.state.gatewayName} activeSmartContract={this.state.activeSmartContract} smartContracts={this.state.smartContracts}
                                switchSmartContract={this.switchSmartContract} postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                        <Route exact path='/transaction/create' render={(): JSX.Element =>
                            <TransactionCreate activeSmartContract={this.state.activeSmartContract} transactionOutput={this.state.transactionOutput}
                                postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                    </div>
                </Router>
            );
        }
    }
}

export default App;
