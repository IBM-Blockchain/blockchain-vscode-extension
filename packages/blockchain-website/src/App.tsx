import React, { Component } from 'react';
import './App.scss';
import StatusList from './components/StatusList/StatusList';
import UpcomingList from './components/UpcomingList/UpcomingList';
import CurrentVersion from './components/CurrentVersion/CurrentVersion';
import HighLevelStatus from './components/HighLevelStatus/HighLevelStatus';

class App extends Component {
    render(): JSX.Element {
        return (
            <div className='header-bar'>
                <p>
                    <span className='ibm'>IBM</span>
                    <span className='page-title'>Blockchain Platform Extension for VS Code</span>
                </p>

                <div className='app-container'>
                    <div className='heading-container'>
                        <p className='status'>Status</p>
                        <p className='title'>IBM Blockchain Platform Extension for VS Code</p>
                        <CurrentVersion/>
                        <HighLevelStatus/>
                        <p className='description'>Check this page for information on any current incidents effecting the latest version of the IBM Blockchain Platform extension for Visual Studio Code, as well as planned content for the next release.</p>
                    </div>
                    <div className='issues-container'>
                        <StatusList></StatusList>
                    </div>

                    <div className='issues-container'>
                        <UpcomingList></UpcomingList>
                    </div>
                </div>
            </div>
        );
    }
}

export default App;
