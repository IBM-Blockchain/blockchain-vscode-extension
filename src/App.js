import React, { Component } from 'react';
import logo from './icon_light.svg';
import StatusList from './components/StatusList';
import UpcomingList from './components/UpcomingList';
import './App.css';

class App extends Component {
  render() {
    return (
        <div className="container">
            <div className="row">
                <div className="col-md-15 col-md-offset-2">
                    <h1 className="text-muted"><img className="logo" src={logo} alt='IBM Blockchain Logo'></img> IBM Blockchain Platform - Status Page</h1>
                    <hr />
                    <p className="lead">
                        Below you can find the latest incidents as well as the information for our next release.
                    </p>
                    <StatusList></StatusList>
                    <hr/>
                    <UpcomingList></UpcomingList>
                </div>
            </div>
        </div>
        )
    }
}

export default App;
