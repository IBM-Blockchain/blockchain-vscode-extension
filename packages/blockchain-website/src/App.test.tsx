import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import chai from 'chai';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('App', () => {
    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<App />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
