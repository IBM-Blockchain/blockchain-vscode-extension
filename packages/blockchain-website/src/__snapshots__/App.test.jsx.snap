// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the expected snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div>
      <h1
        className="text-muted"
      >
        <img
          alt="IBM Blockchain Logo"
          className="logo"
          src="ext_logo.svg"
        />
         IBM Blockchain Platform Extension for VSCode - Status Page
      </h1>
      <hr />
      <h2
        className="text-muted"
      >
         
        <div />
         
      </h2>
      <p
        className="lead"
      >
        Check this page for information on any current incidents effecting the latest version of the IBM Blockchain Platform extension for Visual Studio Code, as well as planned content for the next release.
      </p>
      <hr />
    </div>
  </div>
</div>
`;
